### DISCUSSION TITLE
What is SMP

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23575] [Q&A General] [answered]
Hello All,
I have some questions. The official documentation states that single matrix preprocessing (SMP) builds a preprocessing using the non-diagonal portion of a user-defined Jacobian matrix. Why is it constructed like this? What's the use of that? When is it appropriate to use SMPS?
Thanks 
![633faea5215d3e380c612b96930d891](https://user-images.githubusercontent.com/126685008/222309365-9b758fd8-6600-48dd-b11b-139bae444872.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

The effects are going to depend on the solve type, PJFNK or NEWTON, but essentially if you ignore the non diagonal part of the Jacobian (also used for preconditioning), you are ignoring the effects of the coupling between variables in some capacity.

- For Newton this likely means non linear iterations wont converge as the Jacobian generally needs to be exact
- For PJFNK, this is only going to affect the preconditioning, and the effect will depend on which type of preconditioning you are using

Guillaume

**Reply by nanguaxiaofendui** [reply]
hello
Thank you very much for your answer. I have one more question.
Why is the lower left element of the M matrix like this?

![微信图片_20230302142530](https://user-images.githubusercontent.com/126685008/222348364-e7292053-d5c1-4be3-87b4-ee31a2688690.png)

**Reply by GiudGiud** [reply]
This is specific to this coupled diffusion problem in the example it s used for

**Reply by nanguaxiaofendui** [reply]
Thank you very much!
